Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2945,24 +2945,8 @@ extension BrowserViewController: IntroViewControllerDelegate {
 }
 
 extension BrowserViewController: FxAContentViewControllerDelegate {
-    func contentViewControllerDidSignIn(_ viewController: FxAContentViewController, data: JSON) {
```

Comment:
The current, somewhat-broken FxA flow does the following:

1. User signs into their FxA account
2. If they've logged in from this IP recently - good to go, account if ready for syncing
3. Otherwise, the user is shown a "Confirm Email" screen.

The problem is that when the user sees that confirm email screen, the FxA server immediately send back a token with the account information. The account is set in the `needsVerification` state instead of the `none` state. This is the cause of the following bug https://bugzilla.mozilla.org/show_bug.cgi?id=1345597.

With that being said, I'm curious about the following scenarios with this code change:
- When a user requires email confirmation, does the FxA screen immediately after it registers with the push service? 
- When the user requires email confirmation, are we still able to successfully register the device for push even though we haven't actually confirmed yet?